ledgerjs-fio/src/interactions/signTransaction.ts

```diff
@@ -0,0 +1,133 @@
+import type {HexString, ParsedTransaction, ParsedTransferFIOTokensData, Uint8_t, ValidBIP32Path} from "types/internal"
+
+import {InvalidDataReason} from "../errors"
+import type {SignedTransactionData, Version} from "../types/public"
+import {HARDENED, Transaction} from "../types/public"
+import {chunkBy} from "../utils/ioHelpers"
+import {validate} from "../utils/parse"
+import {uint8_to_buf} from "../utils/serialize"
+import {
+    buf_to_hex,
+    date_to_buf,
+    hex_to_buf,
+    path_to_buf,
+    uint16_to_buf,
+    uint32_to_buf,
+    uint64_to_buf,
+} from "../utils/serialize"
+import {INS} from "./common/ins"
+import type {Interaction, SendParams} from "./common/types"
+import {ensureLedgerAppVersionCompatible} from "./getVersion"
+
+const enum P1 {
+    STAGE_INIT = 0x01,
+    STAGE_HEADER = 0x02,
+    STAGE_ACTION_HEADER = 0x03,
+    STAGE_ACTION_AUTHORIZATION = 0x04,
+    STAGE_ACTION_DATA = 0x05,
+    STAGE_WITNESSES = 0x10,
+}
+
+const send = (params: {
+    p1: number,
+    p2: number,
+    data: Buffer,
+    expectedResponseLength?: number
+}): SendParams => ({ins: INS.SIGN_TX, ...params})
+
+
+export function* signTransaction(version: Version, parsedPath: ValidBIP32Path, chainId: HexString, tx: ParsedTransaction): Interaction<SignedTransactionData> {
+    ensureLedgerAppVersionCompatible(version)
+
+    //Initialize and send chainId
+    {
+        const P2_UNUSED = 0x00
+        const response = yield send({
```

Comment:
This is a repeating thing in this interaction, but you assign something to `const response`, but you are actually not using it.
```suggestion
        yield send({
```